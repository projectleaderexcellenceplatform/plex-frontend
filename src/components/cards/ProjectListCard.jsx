import React, {useState} from "react";
import avtar from "../../assets/images/avtar.svg";
import blueDelete from "../../assets/images/icons/blueDelete.svg";
import group from "../../assets/images/icons/group.svg";

const ProjectListCard = (props) => {
    const [active, setActive] = useState(true);
    const project = props.project;

    return (
        <div data-testid={"projectlistcard"}>
            {project
                .sort((a, b) => a.id > b.id ? 1 : -1)
                .map((item) => {
                    return (
                        <div key={item.id}
                             className={"project-Card-b3 justify-content-between d-flex w-100 rounded-10 ps-3"}>
                            <div className={"row align-items-center"}>
                                <div className={"col-10"}>
                                    <div className={"mb-0 f16 color1 fw600"}>
                                        {item.title}
                                    </div>
                                </div>
                            </div>

                            {item.categories.map(cat => (
                                <div key={cat.id} className={"d-flex"}>
                                    <div className={"align-self-center pe-4"}>
                                        <div className={"pe-4"}>
                                            <div className={"d-flex"}>
                                                <div className={"bg-color4 px-2 rounded-3 f14 fw600 color1 ms-4"}>
                                                    {cat.name}
                                                </div>
                                            </div>
                                        </div>
                                    </div>
                                    <div
                                         className={"list-card-delete align-self-center h-100 px-2 bg-color4 py-4 pointer"}>
                                        <img src={blueDelete} alt={""}/>
                                    </div>
                                </div>
                            ))}

                        </div>
                    )
                })}


        </div>


    )
}

export default ProjectListCard;
