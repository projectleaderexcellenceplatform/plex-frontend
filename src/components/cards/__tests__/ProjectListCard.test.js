import React, {useState} from "react";

import {render, screen, cleanup} from "@testing-library/react";
import renderer from 'react-test-renderer'
import ProjectListCard from "../ProjectListCard";

afterEach(() => {
    cleanup();
})

test('test render', () => {
    const project = [
        {
            "id": 1,
            "title": "Project1",
            "shortDescription": "Omschrijving1",
            "uploader": "uploader1",
            "categories": [
                {
                    "id": 1,
                    "name": "UX/UI"
                }
            ]
        },
        {
            "id": 7,
            "title": "Project7",
            "shortDescription": "Omschrijving7",
            "uploader": "uploader7",
            "categories": []
        }
    ]

    console.log(project)
    render(<ProjectListCard project={project}/>)
    const projectListCardElement = screen.getByTestId('projectlistcard')
    expect(projectListCardElement).toBeInTheDocument()
})

test('matches snapshot', () => {
    const project = [
        {
            "categories": [
                {
                    "id": 1,
                    "name": "UX/UI"
                }
            ],
            "projects": [
                {
                    "id": 7,
                    "title": "Project7",
                    "shortDescription": "Omschrijving7",
                    "uploader": "uploader7",
                    "categories": [
                        {
                            "id": 1,
                            "name": "UX/UI"
                        }
                    ]
                },
                {
                    "id": 1,
                    "title": "Project1",
                    "shortDescription": "Omschrijving1",
                    "uploader": "uploader1",
                    "categories": [
                        {
                            "id": 1,
                            "name": "UX/UI"
                        }
                    ]
                }
            ]
        }
    ]

    const tree = renderer.create(<ProjectListCard project={project}/>).toJSON();
    expect(tree).toMatchSnapshot();
});