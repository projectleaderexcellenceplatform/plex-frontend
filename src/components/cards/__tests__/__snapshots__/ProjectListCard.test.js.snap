// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="project-Card-b3 justify-content-between d-flex w-100 rounded-10 ps-3"
  >
    <div
      className="row align-items-center"
    >
      <div
        className="col-10"
      >
        <div
          className="mb-0 f16 color1 fw600"
        />
      </div>
    </div>
    <div
      className="d-flex"
    >
      <div
        className="align-self-center pe-4"
      >
        <div
          className="pe-4"
        >
          <div
            className="d-flex"
          >
            <div
              className="bg-color4 px-2 rounded-3 f14 fw600 color1 ms-4"
            >
              UX/UI
            </div>
          </div>
        </div>
      </div>
      <div
        className="list-card-delete align-self-center h-100 px-2 bg-color4 py-4 pointer"
      >
        <img
          alt=""
          src="blueDelete.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
